### PR TITLE
Gci 1210 require forename surname for collection

### PR DESCRIPTION
**Pull request by rjadhav-ch**
This adds validators to ensure that forename and surname are set when delivery type is collection.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/controller/CertifiedCopiesItemController.java [hunk 1]
```diff
@@ -29,30 +30,40 @@
...
             HttpServletRequest request,
             final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId) {
 
         Map<String, Object> logMap = LoggingUtils.createLoggingDataMap(requestId);
         LoggingUtils.getLogger().infoRequest(request, "create certified copy item request", logMap);
 
+        final List<String> errors = createItemRequestValidator.getValidationErrors(certifiedCopyItemRequestDTO);
+        if (!errors.isEmpty()) {
+            logErrorsWithStatus(logMap, errors, BAD_REQUEST);
+            LOGGER.errorRequest(request, "create certificate item validation errors", logMap);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/controller/CertifiedCopiesItemController.java:63]: `Create certified copy item...`

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateItemRequestValidator.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import uk.gov.companieshouse.certifiedcopies.orders.api.util.FieldNameConverter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implements validation of the request payload specific to the the create item request only.
+ */
+@Component
+public class CreateItemRequestValidator extends RequestValidator {
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateItemRequestValidator.java:17]: `CreateItemRequestValidator`. Given this validates certified copy items, I would give it a more specific name.

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/controller/CertifiedCopiesItemController.java (hunk 1 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/controller/CertifiedCopiesItemController.java]: oops. Thanks. This is fixed now.

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateItemRequestValidator.java (hunk 2 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateItemRequestValidator.java]: This has been renamed.

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        final List<String> errors = new ArrayList<>();
+        if (options.getDeliveryMethod() == DeliveryMethod.COLLECTION) {
+            if (isBlank(options.getForename())) {
+                errors.add("forename: must not be blank when delivery method is collection");
+            }
+            if (isBlank(options.getSurname())) {
+                errors.add("surname: must not be blank when delivery method is collection");
+            }
+        }
+        return errors;
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java:59]: This should have logging in it? If the logging of the errors is being handled elsewhere then the LOGGER should be removed if it is not being used

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.Boolean.TRUE;
+import static java.util.Arrays.stream;
+import static org.apache.commons.lang.StringUtils.isBlank;
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java:19]: Sonar report indicates some of these imports are unused. Can the unused imports be removed please?

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateCertifiedCopyItemRequestValidator.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.certifiedcopies.orders.api.dto.CertifiedCopyItemOptionsRequestDTO;
+import uk.gov.companieshouse.certifiedcopies.orders.api.dto.CertifiedCopyItemRequestDTO;
+import uk.gov.companieshouse.certifiedcopies.orders.api.model.CertifiedCopyItem;
+import uk.gov.companieshouse.certifiedcopies.orders.api.model.CertifiedCopyItemOptions;
+import uk.gov.companieshouse.certifiedcopies.orders.api.util.FieldNameConverter;
+
+import java.util.ArrayList;
+import java.util.List;
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateCertifiedCopyItemRequestValidator.java:11]: Sonar report indicates unused imports. Can the unused imports be removed please?

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java (hunk 3 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java:59]: The validators validates, collects errors and returns it to the controller where the logging is handled. There is no LOGGER in use in this validator.

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateCertifiedCopyItemRequestValidator.java (hunk 5 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/CreateCertifiedCopyItemRequestValidator.java]: These have been removed.

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java (hunk 4 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java:19]: These have been removed.

File: src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Implements common request payload validation.
+ */
+public class RequestValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingUtils.APPLICATION_NAMESPACE);
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/certifiedcopies/orders/api/validator/RequestValidator.java:26]: If the LOGGER isn't used it should be removed (along with any associated import)

**Pull request merged**